controllers/controller_test.go

```diff
@@ -354,6 +364,32 @@ func verifyFinalizerExists() {
 	ExpectWithOffset(1, controllerutil.ContainsFinalizer(snr, controllers.SNRFinalizer)).Should(BeTrue(), "finalizer should be added")
 }
 
+func verifyNoExecuteTaintRemoved() {
+	By("Verify that node does not have NoExecute taint")
+	Eventually(isNoExecuteTaintExist(), 10*time.Second, 200*time.Millisecond).Should(BeFalse())
+}
+
+func verifyNoExecuteTaintExist() {
+	By("Verify that node has NoExecute taint")
+	Eventually(isNoExecuteTaintExist(), 10*time.Second, 200*time.Millisecond).Should(BeTrue())
+}
+
+func isNoExecuteTaintExist() func() bool {
```

Comment:
this can be made easier: no need to return a new func(), as this already is a func. You just need to pass in this func just by name without () in the Eventually above:

```
... Eventually(isNoExecuteTaintExist,...

func isNoExecuteTaintExist() (bool, error) { // no new func here...}
```

Reply:
I might be inclined to have the value be the operator name, eg. `self-node-remediation`

Reply:
so `medik8s.io/remediation=self-node-remediation:NoExecute` ?
Works for me 👍🏼